src/main/java/net/dries007/tfc/objects/items/ItemsTFC.java

```diff
@@ -187,7 +193,7 @@ public static void registerItems(RegistryEvent.Register<Item> event)
         {
             for (Metal metal : TFCRegistries.METALS.getValuesCollection())
             {
-                if (type.hasType(metal))
+                if (type != Metal.ItemType.BUCKET && type.hasType(metal))
```

Comment:
```suggestion
                if (type.hasType(metal))
```
The modification to `Metal.ItemType#hasType` should already cover this

Reply:
The modification of hasType ensures that no other metals are valid buckets (wrought_iron, bronzes, etc.). This modification ensures that buckets are not registered via the simple registry to handle their custom filled fluid models.

Thanks for feedback!

Reply:
I see the problem. This makes sense then.